tests/routes/test_analysis.py

```diff
@@ -0,0 +1,188 @@
+import pytest
+
+from app.errors import InvalidResponseError
+from tests import AWSMock
+
+TEST_FUNCTION_NAME = "test_raster_analysis"
+
+
+@pytest.fixture(scope="session", autouse=True)
+def analysis_lambda_client():
+    services = ["lambda", "iam", "logs"]
+    aws_mock = AWSMock(*services)
+
+    resp = aws_mock.mocked_services["iam"]["client"].create_role(
+        RoleName="TestRole", AssumeRolePolicyDocument="some_policy"
+    )
+    iam_arn = resp["Role"]["Arn"]
+
+    aws_mock.create_lambda_function(
+        FUNC_STR, TEST_FUNCTION_NAME, "lambda_function.lambda_handler", "python3.7", iam_arn
+    )
+
+    yield aws_mock.mocked_services["lambda"]["client"]
+
+    aws_mock.stop_services()
+
+
+@pytest.mark.asyncio
+async def test_raster_analysis__success(async_client, analysis_lambda_client):
+    """Basic test to check if lambda is successfully called and returns response correctly."""
+    response = await async_client.get(
+        f"/analysis/zonal/{SAMPLE_GEOSTORE_ID}?geostore_origin=rw&group_by=umd_tree_cover_loss__year&filters=is__umd_regional_primary_forest_2001&filters=umd_tree_cover_density_2000__30&sum=area__ha"
+    )
+
+    assert response.status_code == 200
+    assert response.json()["status"] == "success"
+    assert response.json()["data"] == SAMPLE_RESPONSE
```

Comment:
Am I crazy or is it a problem that SAMPLE_GEOSTORE_ID and SAMPLE_RESPONSE are not defined until later? Did you re-arrange things after testing? Or maybe I just forget how Python works...

Reply:
In Python this works because it'll read all the top level code when the module is loaded, and then run pytest. So it reads the function definition but doesn't actually execute it, then reads the constants, then executes the pytest functions